src/workerd/jsg/jsvalue.c++

```diff
@@ -300,7 +301,40 @@ size_t JsString::utf8Length(jsg::Lock& js) const {
 }
 
 kj::String JsString::toString(jsg::Lock& js) const {
-  return kj::str(inner);
+  return js.accountedKjString(kj::str(inner));
+}
+
+jsg::USVString JsString::toUSVString(Lock& js) const KJ_WARN_UNUSED_RESULT {
+  auto buf = kj::heapArray<char>(inner->Utf8LengthV2(js.v8Isolate) + 1);
+  inner->WriteUtf8V2(js.v8Isolate, buf.begin(), buf.size(),
+      v8::String::WriteFlags::kNullTerminate | v8::String::WriteFlags::kReplaceInvalidUtf8);
+  return js.accountedUSVString(kj::mv(buf));
+}
+
+jsg::ByteString JsString::toByteString(Lock& js) const KJ_WARN_UNUSED_RESULT {
+  jsg::ByteString result(toString(js));
+
+  if (!simdutf::validate_ascii(result.begin(), result.size())) {
```

Comment:
We're duplicating the code here. Can we avoid having multiple - exact methods?

Reply:
I was just working on consolidating dups in a follow on commit